ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/network/DefaultP2PNetwork.java

```diff
@@ -394,12 +387,11 @@ public boolean removeMaintainedConnectionPeer(final Peer peer) {
     return removed;
   }
 
-  public void checkMaintainedConnectionPeers() {
-    for (final Peer peer : peerMaintainConnectionList) {
-      if (!(isConnecting(peer) || isConnected(peer))) {
-        connect(peer);
-      }
-    }
+  void checkMaintainedConnectionPeers() {
+    peerMaintainConnectionList.stream()
+        .filter(p -> !isConnectingOrConnected(p))
+        .filter(this::isPeerAllowed)
```

Comment:
Check permissions before connecting :\

Reply:
Shouldn't we drop them from the maintained list if they aren't permitted?

Reply:
I don't think so. The peer might later be permitted.